dashboard/app/models/course.rb

```diff
@@ -387,6 +387,18 @@ def self.course_cache
   end
 
   def self.get_without_cache(id_or_name)
+    # When the caller requests csp-2017, make sure we serve the CSP 2017 course,
+    # regardless of whether it has been renamed from csp to csp-2017 yet.
+    if id_or_name == 'csp-2017'
+      return Course.where(name: ['csp', 'csp-2017']).first
```

Comment:
[Subset conditions](http://guides.rubyonrails.org/active_record_querying.html#subset-conditions) - equivalent to `WHERE (name IN ('csp', 'csp-2017'))` - TIL!